fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/identifiers/FedoraID.java

```diff
@@ -0,0 +1,197 @@
+     */
+    public boolean isMemento() {
+        return isMemento;
+    }
+
+    /**
+     * Is the identifier for an ACL?
+     * @return true if the id is for the fcr:acl endpoint.
+     */
+    public boolean isAcl() {
+        return isAcl;
+    }
+
+    /**
+     * Is the identifier for a timemap?
+     * @return true if id for the fcr:versions endpoint and NOT a memento.
+     */
+    public boolean isTimemap() {
+        return isTimemap;
+    }
+
+    /**
+     * Is the identifier for a nonRdfSourceDescription?
+     * @return true if id for the fcr:metadata endpoint
+     */
+    public boolean isDescription() {
+        return isNonRdfSourceDescription;
+    }
+
+    /**
+     * Is the identifier for a hash uri?
+     * @return true if full id referenced a hash uri.
+     */
+    public boolean isHashUri() {
+        return hashUri != null;
+    }
+
+    /**
+     * Get the hash uri.
+     * @return the hash uri from the id or null if none.
+     */
+    public String getHashUri() {
+        return hashUri;
+    }
+
+    /**
+     * Return the ID of the base resource for this request.
+     * @return the shorten id.
+     */
+    public String getResourceId() {
+        return id;
+    }
+
+    /**
+     * Return the original full ID.
+     * @return the id.
+     */
+    public String getFullId() {
+        return fullId;
+    }
+
+    /**
+     * Return the Memento datetime as Instant.
+     * @return The datetime or null if not a memento.
+     */
+    public Instant getMementoInstant() {
+        return mementoDatetime;
+    }
+
+    /**
+     * Return the Memento datetime string.
+     * @return The yyyymmddhhiiss memento datetime or null if not a Memento.
+     */
+    public String getMementoString() {
+        return mementoDatetime_str;
+    }
+
+    /**
+     * Process the original ID into its parts without using a regular expression.
+     */
+    private void processIdentifier() {
+        // Regex pattern which decomposes a http resource uri into components
+        // The first group determines if it is an fcr:metadata non-rdf source.
+        // The second group determines if the path is for a memento or timemap.
+        // The third group allows for a memento identifier.
+        // The fourth group for allows ACL.
+        // The fifth group allows for any hashed suffixes.
+        // ".*?(/" + FCR_METADATA + ")?(/" + FCR_VERSIONS + "(/\\d{14})?)?(/" + FCR_ACL + ")?(\\#\\S+)?$");
+        String processID = this.fullId;
+        if (processID.equals(FEDORA_ID_PREFIX)) {
+            this.isRepositoryRoot = true;
+            this.id = this.fullId;
+            // Root has no other possible endpoints, so short circuit out.
+            return;
+        }
+        if (processID.contains("#")) {
+            this.hashUri = processID.split("#")[1];
+            processID = processID.split("#")[0];
+        }
+        if (processID.contains(FCR_ACL)) {
```

Comment:
What if the id looks like `a/b/fcr:acl/garbage`? Should that be an error rather than simply throwing out the invalid content?

Reply:
Fedora 5.1 responds with:
```
> curl -ufedoraAdmin:fedoraAdmin http://localhost:8080/rest/A/fcr:acl/test-garbage -i
HTTP/1.1 400 Bad Request
Date: Mon, 23 Mar 2020 15:53:58 GMT
Set-Cookie: JSESSIONID=67t4kxw6llqe1gk3lq8rgj6gb;Path=/
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Set-Cookie: rememberMe=deleteMe; Path=/; Max-Age=0; Expires=Sun, 22-Mar-2020 15:53:58 GMT
Content-Type: text/plain;charset=utf-8
Content-Length: 48
Server: Jetty(9.3.25.v20180904)

Error converting "fcr:acl" from String to a Name
```
What do we want to see with Fedora 6?

Reply:
`400` makes sense. Just about any error message would be better than the 5.1 message.

Reply:
Resolved with 608e69dd845edb56d36941606048ebcfa493634f